scrapy/core/engine.py

```diff
@@ -26,7 +28,10 @@ class Slot:
     def __init__(self, start_requests, close_if_idle, nextcall, scheduler):
         self.closing = False
         self.inprogress = set()  # requests in progress
-        self.start_requests = iter(start_requests)
+        if hasattr(inspect, 'isasyncgen') and inspect.isasyncgen(start_requests):
```

Comment:
what do you think about adding a helper function with this code? It is used several times in the codebase, and used in docs as well.

Reply:
It's only needed while we support 3.5 but I agree it can be useful.